### PR TITLE
Handle `_apply_iterate` handling, but disable overdubbing of `iterf` in `_apply_iterate`

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

File: src/overdub.jl [hunk 1]
```diff
@@ -39,7 +39,9 @@ else
     specialize_method(method, metharg, methsp, world, force) = Core.Compiler.specialize_method(method, metharg, methsp, force)
 end
 
-
+function overdubbed_iterate(ctx, iterate)
+    (args...) -> overdub(ctx, iterate)
+end
```
Comment (MasonProtter) [inline src/overdub.jl:44]: I think this needs to take (and be passed) additional args, e.g. 
```julia
function overdubbed_iterate(ctx, iterate, args...)
    (args...) -> overdub(ctx, iterate, args...)
end
```

Comment (MasonProtter) [reply src/overdub.jl:44]: But doing that causes test failuse elsewhere, (VATupleCtx test) but it makes all the Nuke tests pass.

**Comment by Roger-luo** [comment]
@oxinabox I feel this solves  https://github.com/oxinabox/AutoPreallocation.jl/issues/17 ? it seems `apply_iterate` was not handled correctly there?

**Comment by vchuravy** [comment]
Yeah the VACtx test stumped me. So I disabled overdubbing of `iterf` for now. There are still two other regressions, but at least this should unbreak Cassette on 1.4

**Comment by KristofferC** [comment]
Wasn't the whole point of introducing `apply_iterate` to be able to overdub the `iterate` in `Core.apply`?

**Comment by vchuravy** [comment]
Yes, this PR doesn't address that, but it correctly deals with `apply_iterate` in all other situations and returns the status quo for Cassette to 1.4.

Activating the overdubbing of `iterate` breaks more things so that requires another look.

**Pull request merged**